bin/light-base/src/sync_service/standalone.rs

```diff
@@ -389,307 +433,290 @@ struct Task<TPlat: Platform> {
+            None => return false,
+        };
 
-            // Before inserting the request back to the syncing state machine, clamp the number
-            // of blocks to the number of blocks we expect to receive.
-            // This constant corresponds to the maximum number of blocks that nodes will answer
-            // in one request. If this constant happens to be inaccurate, everything will still
-            // work but less efficiently.
-            request_detail.num_blocks_clamp(NonZeroU64::new(64).unwrap());
-
-            match request_detail {
-                all::RequestDetail::BlocksRequest {
-                    first_block_hash,
-                    first_block_height,
-                    ascending,
-                    num_blocks,
-                    request_headers,
-                    request_bodies,
-                    request_justification,
-                } => {
-                    let peer_id = self.sync[source_id].0.clone(); // TODO: why does this require cloning? weird borrow chk issue
-
-                    let block_request = self.network_service.clone().blocks_request(
-                        peer_id,
-                        self.network_chain_index,
-                        network::protocol::BlocksRequestConfig {
-                            start: if let Some(first_block_hash) = first_block_hash {
-                                network::protocol::BlocksRequestConfigStart::Hash(first_block_hash)
-                            } else {
-                                network::protocol::BlocksRequestConfigStart::Number(
-                                    first_block_height,
-                                )
-                            },
-                            desired_count: NonZeroU32::new(
-                                u32::try_from(num_blocks.get()).unwrap_or(u32::max_value()),
-                            )
-                            .unwrap(),
-                            direction: if ascending {
-                                network::protocol::BlocksRequestDirection::Ascending
-                            } else {
-                                network::protocol::BlocksRequestDirection::Descending
-                            },
-                            fields: network::protocol::BlocksRequestFields {
-                                header: request_headers,
-                                body: request_bodies,
-                                justifications: request_justification,
-                            },
+        // Before inserting the request back to the syncing state machine, clamp the number
+        // of blocks to the number of blocks we expect to receive.
+        // This constant corresponds to the maximum number of blocks that nodes will answer
+        // in one request. If this constant happens to be inaccurate, everything will still
+        // work but less efficiently.
+        request_detail.num_blocks_clamp(NonZeroU64::new(64).unwrap());
```

Comment:
can we extract this into a const?

Reply:
Same remark. I've just changed the indentation level.